integration/validation_separate_metrics_test.go

```diff
@@ -0,0 +1,268 @@
+
+package integration
+
+import (
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/grafana/e2e"
+	e2edb "github.com/grafana/e2e/db"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/mimir/integration/e2emimir"
+)
+
+func TestValidateSeparateMetrics(t *testing.T) {
+	tests := map[string]struct {
+		labelToSearch string
+		configFlagSet bool
+		metricExists  bool
+	}{
+		"No separate metrics label present": {
+			labelToSearch: "",
+			configFlagSet: false,
+			metricExists:  true,
+		},
+		"Check for correct label": {
+			labelToSearch: "test-group",
+			configFlagSet: true,
+			metricExists:  true,
+		},
+		"Check for incorrect label": {
+			labelToSearch: "incorrect-group",
+			configFlagSet: true,
+			metricExists:  false,
+		},
+	}
+
+	for testName, testData := range tests {
+		t.Run(testName, func(t *testing.T) {
+			s, err := e2e.NewScenario(networkName)
+			require.NoError(t, err)
+			defer s.Close()
+
+			flags := mergeFlags(
+				BlocksStorageFlags(),
+				BlocksStorageS3Flags(),
+			)
+
+			if testData.configFlagSet {
+				flags["-validation.separate-metrics-label"] = "group_1"
+			}
+
+			// Start dependencies.
+			consul := e2edb.NewConsul()
+			minio := e2edb.NewMinio(9000, flags["-blocks-storage.s3.bucket-name"])
+			require.NoError(t, s.StartAndWaitReady(consul, minio))
+
+			// Start Mimir components.
+			distributor := e2emimir.NewDistributor("distributor", consul.NetworkHTTPEndpoint(), flags)
+			ingester1 := e2emimir.NewIngester("ingester-1", consul.NetworkHTTPEndpoint(), flags)
+			ingester2 := e2emimir.NewIngester("ingester-2", consul.NetworkHTTPEndpoint(), flags)
+			ingester3 := e2emimir.NewIngester("ingester-3", consul.NetworkHTTPEndpoint(), flags)
+			require.NoError(t, s.StartAndWaitReady(distributor, ingester1, ingester2, ingester3))
+
+			// Wait until distributor has updated the ring.
+			require.NoError(t, distributor.WaitSumMetricsWithOptions(e2e.Equals(3), []string{"cortex_ring_members"}, e2e.WithLabelMatchers(
+				labels.MustNewMatcher(labels.MatchEqual, "name", "ingester"),
+				labels.MustNewMatcher(labels.MatchEqual, "state", "ACTIVE"))))
+
+			// Wait until ingesters have heartbeated the ring after all ingesters were active,
+			// in order to update the number of instances. Since we have no metric, we have to
+			// rely on a ugly sleep.
+			time.Sleep(2 * time.Second)
+
+			now := time.Now()
+			client, err := e2emimir.NewClient(distributor.HTTPEndpoint(), "", "", "", userID)
+			require.NoError(t, err)
+
+			// Push an invalid metric to increment cortex_discarded_samples_total
+			series, _, _ := generateSeries("TestMetric", now, prompb.Label{
+				Name:  "Test|Invalid|Label|Char",
+				Value: "123",
+			}, prompb.Label{
+				Name:  "group_1",
+				Value: "test-group",
+			})
+
+			res, err := client.Push(series)
+			require.NoError(t, err)
+			require.Equal(t, 400, res.StatusCode)
```

Comment:
Can we loop-push here multiple requests with:
- This testcase
- No group label
- No group label in first series, but group label in the second series
- Different group labels

There's no need to restart the whole test setup, we can just push all of them and then check their respective metrics.
-